crates/holochain_state/src/mutations.rs

```diff
@@ -8,19 +8,84 @@ use holo_hash::encode::blake2b_256;
 use holo_hash::*;
 use holochain_nonce::Nonce256Bits;
 use holochain_sqlite::prelude::DatabaseResult;
+use holochain_sqlite::rusqlite;
 use holochain_sqlite::rusqlite::named_params;
 use holochain_sqlite::rusqlite::types::Null;
 use holochain_sqlite::rusqlite::Transaction;
 use holochain_sqlite::sql::sql_conductor;
 use holochain_types::prelude::*;
 use holochain_types::sql::AsSql;
-use kitsune_p2p::dependencies::kitsune_p2p_fetch::TransferMethod;
 use std::str::FromStr;
 
 pub use error::*;
 
 mod error;
 
+/// Gossip has two distinct variants which share a lot of similarities but
+/// are fundamentally different and serve different purposes
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Display,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub enum GossipType {
+    /// The Recent gossip type is aimed at rapidly syncing the most recent
+    /// data. It runs frequently and expects frequent diffs at each round.
+    Recent,
+    /// The Historical gossip type is aimed at comprehensively syncing the
+    /// entire common history of two nodes, filling in gaps in the historical
+    /// data. It runs less frequently, and expects diffs to be infrequent
+    /// at each round.
+    Historical,
+}
+
+/// The possible methods of transferring op hashes
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Display,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub enum TransferMethod {
+    /// Transfer by publishing
+    Publish,
+    /// Transfer by gossiping
+    Gossip(GossipType),
+}
```

Comment:
We'll probably want to change or just delete this entirely... but pulling it out of legacy kitsune for now to not break things.